lib/modules/solidity/solcP.js

```diff
@@ -24,34 +32,59 @@ class SolcProcess extends ProcessWrapper {
     return {error: 'File not found'};
   }
 
-  loadCompiler(solcLocation) {
-    this.solc = require(solcLocation);
+  installAndLoadCompiler(solcVersion, packagePath){
+    let self = this;
+    return new Promise((resolve) => {
+      let manager = new PluginManager({pluginsPath: packagePath});
+      let timer;
+      if (!fs.existsSync(packagePath)) {
+        timer = new NpmTimer({logger: self._logger, packageName: 'solc', version: solcVersion, showSpinner: true});
+      }
+
+      if(timer) timer.start();
+      manager.install('solc', solcVersion).then(() => {
+        self.solc = manager.require('solc');
+        if(timer) timer.end();
+        resolve();
+      });
+      
+    });
   }
 
   compile(jsonObj, cb) {
     // TODO: only available in 0.4.11; need to make versions warn about this
-    let output = this.solc.compileStandardWrapper(JSON.stringify(jsonObj), this.findImports);
-    cb(output);
+    try {
+      let output = this.solc.compileStandardWrapper(JSON.stringify(jsonObj), this.findImports);
```

Comment:
Why does compileStandardWrapper throw?

Reply:
Additionally, any uncaught compilation errors can be returned and handled by embark better

Reply:
Ok, makes sense 👍